### ISSUE TITLE
Copied HTML content through textutil and pbcopy is not showing

### ISSUE THREAD

**Issue by selipequenote** [p0deje/Maccy#474]
I have a use case where I am dynamically pushing html content to the clipboard using a shell command and noticed that maccy for some reason does not support it.

**Steps to reproduce:**
1. Open the terminal
2. Execute the following command
```
echo "<a href='www.google.com'>google</a>" | textutil -stdin -stdout -format html -convert rtf | pbcopy
```
3. Paste into the Notes app and get a clickable link like this: [google](www.google.com)
4. Trigger maccy and the copied content does not appear on the list

**Notes:**

- It doesn't appear to be an issue with pbcopy because plain text content works fine, like this example
```
echo "this is a test" | pbcopy
```

**Comment by selipequenote** [comment]
tried a couple of other options to push html content to the clipboard but didn't find one yet that works with maccy, currently using something similar to the following apple script:

```AppleScript
set rawHTML to "<a href=www.google.com>google</a>"
set escapedData to do shell script "echo " & (quoted form of rawHTML) as «class HTML»
set the clipboard to escapedData
```

**Comment by andrewpeal** [comment]
I located where this was failing in Maccy. In Clipboard() line 177 this condition `if types.contains(.string) && isEmptyString(item)` is true because the string type is empty. The data is there but it is being populated into the rtf type instead of string.

**Comment by p0deje** [comment]
If only `rtf` type is present, Maccy will ignore it indeed as it's unaware of how to present this data inside its UI. I suppose there is a way to convert RTF to a plain string for the UI purpose, so I think this is something that can be fixed.

**Comment by andrewpeal** [comment]
I was able to format the rtf data using `NSAttributedString(rtf: )` as seen in the development branch of my fork. but what I noticed is the rtf field of NSPasteboard is populated quite often. Copying any text in MacOS for example, populates the formatting data into the rtf field. So would need a new approach in Maccy, to review the rtf data and determine whether or not it is relevant to the current copy operation. In this example, which part of the copied data should Maccy capture, the text, the hyperlink, or the full rtf data?

**Comment by p0deje** [comment]
Whenever there is a pasteboard item that has RTF/HTML but no plain text, Maccy should still capture and store it. For UI purposes, it needs to construct a plain text string from the input somehow.

**Comment by andrewpeal** [comment]
Yep, agreed. The plain text is easily extracted from the rtf data using NSAttributedString.

**Comment by p0deje** [comment]
This is a bit trickier that I anticipated, but I've just pushed a first attempt to support RTF/HTML copies. Both examples of `textutil` and AppleScript now work, though copying RTF **from** Maccy would not properly detect the old copy and create a new one. There are probably more edge cases so I'll do some testing.
